test/core/surface/completion_queue_test.cc

```diff
@@ -365,18 +366,31 @@ static void test_callback(void) {
       GRPC_CQ_DEFAULT_POLLING, GRPC_CQ_NON_LISTENING, GRPC_CQ_NON_POLLING};
   grpc_completion_queue_attributes attr;
   unsigned i;
+  static gpr_mu mu, shutdown_mu;
+  static gpr_cv cv, shutdown_cv;
+  static int cb_counter;
+  gpr_mu_init(&mu);
+  gpr_mu_init(&shutdown_mu);
+  gpr_cv_init(&cv);
+  gpr_cv_init(&shutdown_cv);
 
   LOG_TEST("test_callback");
 
+  gpr_mu_lock(&shutdown_mu);
   bool got_shutdown = false;
+  gpr_mu_unlock(&shutdown_mu);
```

Comment:
Are the locks around this initialization of shutdown_mu really needed? There's a happens-before relationship between later uses through the RPC call and executor enqueueing, correct? Generally speaking, locks around an initialization or in  a destructor are a code smell that something else is wrong.

Reply:
Fixed.